x/merkledb/trieview.go

```diff
@@ -655,43 +636,50 @@ func (t *trieView) remove(key Key) error {
 	}
 
 	nodeToDelete.setValue(maybe.Nothing[[]byte]())
-	if len(nodeToDelete.children) != 0 {
-		// merge this node and its child into a single node if possible
-		return t.compressNodePath(parent, nodeToDelete)
-	}
 
 	// if the removed node has no children, the node can be removed from the trie
-	if err := t.recordNodeDeleted(nodeToDelete); err != nil {
-		return err
-	}
-	if parent != nil {
+	if len(nodeToDelete.children) == 0 {
+		if err := t.recordNodeDeleted(nodeToDelete); err != nil {
+			return err
+		}
+
+		if nodeToDelete.key == t.root.Value().key {
```

Comment:
Might want to move this into record node deleted since theoretically nodes get deleted from other functions as well.  We never actually delete the root from these other functions currently, but might be safer for future changes.

Reply:
Nodes are only deleted in this method or its callees, so I'd be alright keeping this here, but I'll move it if you feel strongly

Reply:
nah, not strongly.